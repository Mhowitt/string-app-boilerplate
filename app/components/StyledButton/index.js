import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Button = styled.button`
  background: ${props => (props.type.primary ? 'tomato' : 'white')};
  color: ${props => (props.type.primary ? 'white' : 'tomato')};
  border: 1px solid ${props => (props.type.primary ? null : 'tomato')};
  padding: 5px 10px;
`;
export default class StyledButton extends React.PureComponent {
  render() {
    const { text, onClick, type } = this.props;
    return (
      <Button onClick={onClick} type={type}>
        {text}
      </Button>
    );
  }
}

StyledButton.propTypes = {
  text: PropTypes.string,
  onClick: PropTypes.func,
  type: PropTypes.string,
};
