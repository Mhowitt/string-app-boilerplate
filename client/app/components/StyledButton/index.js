import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Button = styled.button`
  background: ${props => (props.type === 'primary' ? '#ff4625' : 'white')};
  color: ${props => (props.type === 'primary' ? 'white' : '#ff4625')};
  border: ${props =>
    props.type === 'primary' ? '1px solid #ff4625' : '1px solid #ff4625'};
  padding: 10px;
  width: 120px;
  border-radius: 5px;
  transition: width 0.6s linear;

  &:hover {
    width: 150px;

`;

const CloseButton = styled.button`
  background: transparent;
  width: 50px;
  font-weight: bold;
  color: white;
`;
export default class StyledButton extends React.PureComponent {
  render() {
    const { onClick, type } = this.props;
    const ButtonType = this.props.type === 'close' ? CloseButton : Button;
    return (
      <ButtonType onClick={onClick} type={type}>
        {this.props.children}
      </ButtonType>
    );
  }
}

StyledButton.propTypes = {
  onClick: PropTypes.func,
  type: PropTypes.string,
  children: PropTypes.node,
};
