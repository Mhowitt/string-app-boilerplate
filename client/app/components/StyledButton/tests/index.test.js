import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import StyledButton from '../index';

describe('<StyledButton />', () => {
  let renderedComponent;
  it('renders', () => {
    renderedComponent = shallow(<StyledButton />);
  });
  it('matches snapshot', () => {
    const serialized = toJson(renderedComponent);
    expect(serialized).toMatchSnapshot();
  });

  it('renders its children', () => {
    const text = 'Submit';
    renderedComponent = shallow(<StyledButton>{text}</StyledButton>);
    expect(renderedComponent.contains(text)).toEqual(true);
  });

  it('handles clicks', () => {
    const onClickSimulate = jest.fn();
    renderedComponent = shallow(<StyledButton onClick={onClickSimulate} />);
    renderedComponent.find('button').simulate('click');
    expect(onClickSimulate).toHaveBeenCalled();
  });
});
