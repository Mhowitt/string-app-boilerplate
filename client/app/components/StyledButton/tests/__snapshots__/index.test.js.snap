// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledButton /> matches snapshot 1`] = `<StyledButton__Button />`;
